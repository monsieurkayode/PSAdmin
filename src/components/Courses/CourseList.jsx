import React from 'react';
import { arrayOf, shape, func } from 'prop-types';
import CourseListRow from './CourseListRow';

/**
 *
 * @function CourseList
 *
 * @param {object} props
 *
 * @returns {JSX} JSX
 */
const CourseList = ({ courses, deleteCourse }) => (
  <table className="table">
    <thead>
      <tr>
        <th>&nbsp;</th>
        <th>Title</th>
        <th>Author</th>
        <th>Category</th>
        <th>Length</th>
        <th>&nbsp;</th>
      </tr>
    </thead>
    <tbody>
      {courses.map(course => (
        <CourseListRow
          key={course.id}
          {...course}
          deleteCourse={deleteCourse}
        />
      ))}
    </tbody>
  </table>
);

CourseList.propTypes = {
  courses: arrayOf(shape({})).isRequired,
  deleteCourse: func.isRequired
};

export default CourseList;
