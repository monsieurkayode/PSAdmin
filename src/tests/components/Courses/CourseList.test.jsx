import React from 'react';
import { shallow } from 'enzyme';
import CourseList from '../../../components/Courses/CourseList';
import mockCourses from '../../__mocks__/mockCourses';

const setup = () => {
  const props = {
    courses: mockCourses,
    deleteCourse: jest.fn()
  };

  return shallow(<CourseList {...props} />);
};

describe('CourseList Component', () => {
  const wrapper = setup();

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a table with a table row having six table header cells', () => {
    expect(wrapper.find('table').length).toBe(1);
    expect(wrapper.find('th').length).toBe(6);
  });

  it('has table headings Title, Author, Category and Length', () => {
    expect(wrapper.find('tr').childAt(1).text()).toEqual('Title');
    expect(wrapper.find('tr').childAt(2).text()).toEqual('Author');
    expect(wrapper.find('tr').childAt(3).text()).toEqual('Category');
    expect(wrapper.find('tr').childAt(4).text()).toEqual('Length');
  });

  it('renders 2 CourseListRow components', () => {
    expect(wrapper.find('CourseListRow').length).toEqual(2);
  });
});
