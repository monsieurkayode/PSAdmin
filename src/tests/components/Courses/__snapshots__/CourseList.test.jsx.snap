// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList Component renders correctly 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
         
      </th>
      <th>
        Title
      </th>
      <th>
        Author
      </th>
      <th>
        Category
      </th>
      <th>
        Length
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
  <tbody>
    <CourseListRow
      authorId="cory-house"
      category="JavaScript"
      deleteCourse={[MockFunction]}
      id="react-flux-building-applications"
      key="react-flux-building-applications"
      length="5:08"
      title="Building Applications in React and Flux"
      watchHref="react-flux-building-applications"
    />
    <CourseListRow
      authorId="cory-house"
      category="Software Practices"
      deleteCourse={[MockFunction]}
      id="clean-code"
      key="clean-code"
      length="3:10"
      title="Clean Code: Writing Code for Humans"
      watchHref="writing-clean-code-humans"
    />
  </tbody>
</table>
`;
